dockerfiles/theia/README.md

```diff
@@ -1,33 +1,101 @@
 
-Example:
+To build all the images you can easily run [build.sh](../../build.sh) script located in the repository root.
 
-```shell
-$ ../theia-dev/build.sh
+CI for checking any pull request in this repository runs this script with `--pr` directive
+```bash
+`build.sh --pr`.
+```
+
+**Note**: `--pr` will build only [limited set](../../docker_image_build.include) of docker images.	
+
+To publish docker images use `--push` directive
+```bash
+`build.sh --push`
+```	
+
+## How to build own Che Theia image with Docker
+
+First you need to build `che-theia-dev` image:	
+
+Run in `dockerfiles/theia-dev` dir:	
+
+```bash	
+./build.sh --build-arg:${GITHUB_TOKEN} --tag:next
 ```
 
-After that, You can build Theia image.
+Then in `dockerfiles/theia` run:
 
-Example:
+```bash	
+./build.sh --build-args:${GITHUB_TOKEN},THEIA_VERSION=master --tag:next --branch:master --git-ref:refs\\/heads\\/master
+```
+
+Where `${GITHUB_TOKEN}` is your GitHub API token, it's used for fetching some vscode libraries that placed on GitHub releases. Without that token build will fail.
+
+This script will build new docker image `eclipse/che-theia-dev:next`. The image will contain Che-Theia based on Theia from master branch.
+
+## Build arguments
+
+General command to build Che-Theia with all possible argiments:
 
 ```bash
-$ ./build.sh --build-args:GITHUB_TOKEN=$GITHUB_TOKEN,THEIA_VERSION=0.3.18 --git-ref:refs\\/heads\\/master
+./build.sh --build-args:${GITHUB_TOKEN},THEIA_VERSION=${THEIA_VERSION} --tag:${IMAGE_TAG} --branch:${THEIA_BRANCH} --git-ref:${THEIA_GIT_REFS} --skip-tests
 ```
 
+### GitHub Token
+
+**`${GITHUB_TOKEN}`**
+
+Some Theia dependencies placed on GitHub repository and fetched from the GitHub by using GitHub API. For that you have to generate [new GitHub API token](https://github.com/settings/tokens) and pass it to the script to have proper build.
+
+Parameter `${GITHUB_TOKEN}` is mandatory.
+
+
 ## Theia version
 
-There's a default Theia version set in the script. This version is then injected in all package.jsons.
-You can override `THEIA_VERSION` by exporting the env before running the script
+**`${THEIA_VERSION}`**
+
+It specifies the Theia version. Build script updates all `package.json` files and uses this value as a version for all `'@theia/*'` packages.
+Also this parameter is used for finding patches. All found patches from `'/patches/${THEIA_VERSION}'` directory will be applied to Theia sources before cloning Che-Theia.
+You can set `THEIA_VERSION` environment variable and don't pass it to the script when building.
+
+Parameter `${THEIA_VERSION}` is optional. If it's not specified, the default value `'master'` will be used.
+
+
+## Image Tag
+
+**`${IMAGE_TAG}`**
+
+A newly created docker image will have this version. Passing `'--tag:1.0.0'` to the script will lead to creating `eclipse/che-theia-dev:1.0.0` docker image.
+
+Parameter `${IMAGE_TAG}` is optioal. If it's not specified, the default value `'next'` will be used.
+
+
+## Theia Branch
+
```

Comment:
Finally, when you build theia and che-theia, isn't it the most important option to pass ? Could you put that on the top (before theia_version). Could you also add a note about which version are being used in the CI to produice which tag ?

Reply:
Done. Thanks